koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -881,65 +853,313 @@ def handler(self, src, target, opts=None):
+        except AttributeError:
+            raise koji.BuildError("method %s doesn't exists in osbs" % create_method)
+
+        build_id = build_response.get_build_name()
+        self.logger.debug("OSBS build id: %r", build_id)
+
+        # When builds are cancelled the builder plugin process gets SIGINT and SIGKILL
+        # If osbs has started a build it should get cancelled
+        def sigint_handler(*args, **kwargs):
+            if not build_id:
+                return
+
+            self.logger.warn("Cannot read logs, cancelling build %s", build_id)
+            self.osbs().cancel_build(build_id)
+
+        signal.signal(signal.SIGINT, sigint_handler)
+
+        self.logger.debug("Waiting for osbs build_id: %s to be scheduled.",
+                          build_id)
+        # we need to wait for kubelet to schedule the build, otherwise it's 500
+        self.osbs().wait_for_build_to_get_scheduled(build_id)
+        self.logger.debug("Build was scheduled")
+
+        osbs_logs_dir = self.resultdir()
+        koji.ensuredir(osbs_logs_dir)
+        pid = os.fork()
+        if pid:
+            try:
+                self._incremental_upload_logs(pid)
+            except koji.ActionNotAllowed:
+                pass
+        else:
+            self._osbs = None
+
+            try:
+                self._write_incremental_logs(build_id, osbs_logs_dir)
+            except Exception as error:
+                self.logger.info("Error while saving incremental logs: %s", error)
+                os._exit(1)
+            os._exit(0)
+
+        response = self.osbs().wait_for_build_to_finish(build_id)
+
+        self.logger.debug("OSBS build finished with status: %s. Build "
+                          "response: %s.", response.status,
+                          response.json)
+
+        self.logger.info("Response status: %r", response.is_succeeded())
+
+        if response.is_cancelled():
+            self.session.cancelTask(self.id)
+            raise ContainerCancelled(
+                'Image build was cancelled by OSBS.'
+            )
+
+        elif response.is_failed():
+            error_message = self._get_error_message(response)
+            if error_message:
+                raise ContainerError('Image build failed. %s. OSBS build id: %s' %
+                                     (error_message, build_id))
+            else:
+                raise ContainerError('Image build failed. OSBS build id: %s' %
+                                     build_id)
+
+        repositories = []
+        if response.is_succeeded():
+            repositories = self._get_repositories(response)
+
+        self.logger.info("Image available in the following repositories: %r",
+                         repositories)
+
+        koji_build_id = None
+        if response.is_succeeded():
+            koji_build_id = self._get_koji_build_id(response)
+
+        self.logger.info("Koji content generator build ID: %s", koji_build_id)
```

Comment:
This code is the same as in `buildContainer` except change in one log message.
Can you extract it into method in base class? Like `handle_build_response(build_response)`/`process_build_response(build_response)`

Reply:
ok